pkg/websocket/connection.go

```diff
@@ -0,0 +1,112 @@
+/*
```

Comment:
Add OWNERS for new directory?  Add a test file?

Reply:
Regarding the testfile: I have a question towards that in the initial description. Is mocking the Dialer interface and creating a `WsConnection` interface that i can then mock out the way to go to test this or do you have a better idea?